### PR TITLE
Adds wallet compliance guide

### PR DESCRIPTION
**Pull request by JeffreyDoyle**

### CODE REVIEW

**Comment by chasefleming** [comment]
@JeffreyDoyle Can you link an issue for more context please?

File: docs/wallet-compliance-guide.md [hunk 1]
```diff
@@ -0,0 +1,27 @@
+# Wallet Compliance Guide
+
+This guide contains a set of critera wallets on Flow must meet in order to be accepted and included in Flow's FCL Wallet Discovery.
```
Comment (chasefleming) [inline docs/wallet-compliance-guide.md:3]: Would add a why here. E.g. this is done to ensure that developers and users have a seamless experience across applications, no wallet specific code is required by developers, and users can expect apps to function as expected and support FCL's open standards. 

Would also note that why services are deemed opt-in if they don't meet that criteria.

File: docs/wallet-compliance-guide.md [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+Wallets will be tested against the [FCL Harness](https://github.com/onflow/fcl-next-harness) application. The FCL Harness will be configured to use each wallet, and will be used to carry out **Authentication**, **Authorization** of a transaction, and **User Signature** over some data.
+
+To test your wallet against the [FCL Harness](https://github.com/onflow/fcl-next-harness) application, clone the repo, start the harness, add your wallet to `discovery.wallet` using the input field provided, and ensure your wallet successfully authenticates and executes each mutate and user signing interaction.
+
+Wallet developers are expected to ensure their wallet meets the required functionality by using the FCL Harness to test their wallet. Prior to inclusion in Flow's FCL Discovery, the Flow team will also carry out this testing to ensure functionality. From time to time, the Flow team will carry out testing to ensure continued functionality.
```
Comment (chasefleming) [inline docs/wallet-compliance-guide.md:27]: Would also reiterate they "why" here

**Review by chasefleming — CHANGES_REQUESTED** [review]
Can you also link this from the readme. Probably under "Adding a Wallet to Discovery."

File: docs/wallet-compliance-guide.md [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+At a minimum, wallets on Flow must provide support for FCL's:
+
+- **Authentication** service: https://github.com/onflow/fcl-js/blob/master/packages/fcl/src/wallet-provider-spec/draft-v4.md#authentication-service
+
+- **Authorization** service: https://github.com/onflow/fcl-js/blob/master/packages/fcl/src/wallet-provider-spec/draft-v4.md#authorization-service
+
+- **User Signature** service: https://github.com/onflow/fcl-js/blob/master/packages/fcl/src/wallet-provider-spec/draft-v4.md#user-signature-service
+
+Wallets may additionally choose to support other FCL services (pre-authz, provable authentication etc), but these additional services are not reqiured to be included in Flow's FCL Wallet Descovery.
```
Comment (chasefleming) [inline docs/wallet-compliance-guide.md:17]:
```suggestion
Wallets may additionally choose to support other FCL services (pre-authz, provable authentication etc), but these additional services are not required to be included in Flow's FCL Wallet Discovery.
```

File: docs/wallet-compliance-guide.md [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+
+These standards ensure a high quality user experience on Flow for applications that choose to use FCL Discovery for wallet discovery. 
+
+End users expect that wallets on FCL Discovery function as expected, and application developers that use FCL Discovery depend on it to ensure wallets meet the minimum required functionality. Other opt-in services are not expected to be provided to application developers by wallets, and thus are not requird by this standard.
```
Comment (chasefleming) [inline docs/wallet-compliance-guide.md:21]:
```suggestion
End users expect that wallets on FCL Discovery function as expected, and application developers that use FCL Discovery depend on it to ensure wallets meet the minimum required functionality. Other opt-in services (wallets with a subset of required services or limited functionality) are not expected to be provided to application developers by wallets, and thus are not required by this standard.
```

**Review by chasefleming — CHANGES_REQUESTED** [review]
Also, still need a link from the readme in the "Adding a Wallet to Discovery" section.

File: docs/wallet-compliance-guide.md [hunk 5]
```diff
@@ -16,6 +16,12 @@ At a minimum, wallets on Flow must provide support for FCL's:
 
 Wallets may additionally choose to support other FCL services (pre-authz, provable authentication etc), but these additional services are not reqiured to be included in Flow's FCL Wallet Descovery.
```
Comment (gregsantos) [inline docs/wallet-compliance-guide.md]:
```suggestion
Wallets may choose to support additional FCL services (pre-authz, account-proof, etc.), but these additional services are not required to be included in Flow's FCL Wallet Discovery.
```

File: docs/wallet-compliance-guide.md [hunk 6]
```diff
@@ -16,6 +16,12 @@ At a minimum, wallets on Flow must provide support for FCL's:
...
 
+These standards ensure a high quality user experience on Flow for applications that choose to use FCL Discovery for wallet discovery. 
+
+End users expect that wallets on FCL Discovery function as expected, and application developers that use FCL Discovery depend on it to ensure wallets meet the minimum required functionality. Other opt-in services are not expected to be provided to application developers by wallets, and thus are not requird by this standard.
```
Comment (gregsantos) [inline docs/wallet-compliance-guide.md:21]:
```suggestion
End users and application developers using FCL Discovery should be able to rely on included wallets meeting the minimum required functionality. Other opt-in services are not expected to be provided to application developers by wallets, and thus are not enforced by this standard.
```

File: docs/wallet-compliance-guide.md [hunk 7]
```diff
@@ -16,6 +16,12 @@ At a minimum, wallets on Flow must provide support for FCL's:
...
+
+This acceptance criteria ensures FCL Discovery meets the needs and expectations of the users and developers that depend on it. 
+
 ## Testing
 
 In order to be included in Flow's FCL Wallet Discovery, new wallets must prove their functionality, and continue to prove their functionality on a recurrent basis from time to time. Wallets that fail to prove their functionality may be subject to removal from FCL Discovery until their functionality is restored.
```
Comment (gregsantos) [inline docs/wallet-compliance-guide.md:27]:
```suggestion
In order to be included in Flow's FCL Wallet Discovery, new wallets must prove their functionality, and continue to prove their functionality on a recurrent basis from time to time. Wallets that fail to prove their functionality may be subject to removal from FCL Discovery until the required functionality is restored.
```

**Review by gregsantos — CHANGES_REQUESTED** [review]

**Review by gregsantos — APPROVED** [review]

**Review by chasefleming — APPROVED** [review]

**Pull request merged**